Cannot watch local replay file if the client was not run

**To Reproduce**
1. Make sure the client is completely closed
2. Try to click to local replay file

**Log or error message**
```
java.lang.RuntimeException: Exception in Application init method
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:896)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Failed to execute ApplicationRunner
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:771)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:758)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:310)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:164)
	at com.faforever.client.FafClientApplication.init(FafClientApplication.java:108)
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:825)
	... 2 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.springframework.web.reactive.function.client.WebClient.get()" because "this.webClient" is null
	at com.faforever.client.api.FafApiAccessor.getMany(FafApiAccessor.java:266)
	at com.faforever.client.api.FafApiAccessor.getMany(FafApiAccessor.java:253)
	at com.faforever.client.mod.ModService.getFeaturedMod(ModService.java:466)
	at com.faforever.client.mod.ModService$$FastClassBySpringCGLIB$$aae9cfd3.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
	at org.springframework.cache.interceptor.CacheInterceptor.lambda$invoke$0(CacheInterceptor.java:54)
	at org.springframework.cache.interceptor.CacheAspectSupport.invokeOperation(CacheAspectSupport.java:366)
	at org.springframework.cache.interceptor.CacheAspectSupport.lambda$handleSynchronizedGet$1(CacheAspectSupport.java:447)
	at org.springframework.cache.caffeine.CaffeineCache$LoadFunction.apply(CaffeineCache.java:168)
	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.lambda$computeIfAbsent$2(UnboundedLocalCache.java:239)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.computeIfAbsent(UnboundedLocalCache.java:235)
	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:108)
	at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62)
	at org.springframework.cache.caffeine.CaffeineCache.get(CaffeineCache.java:90)
	at org.springframework.cache.interceptor.CacheAspectSupport.handleSynchronizedGet(CacheAspectSupport.java:442)
	at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:382)
	at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:345)
	at org.springframework.cache.interceptor.CacheInterceptor.invoke(CacheInterceptor.java:64)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:698)
	at com.faforever.client.mod.ModService$$EnhancerBySpringCGLIB$$b87227b0.getFeaturedMod(<generated>)
	at com.faforever.client.game.GameService.runWithReplay(GameService.java:359)
	at com.faforever.client.game.GameService$$FastClassBySpringCGLIB$$eb8a13b.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
	at com.faforever.client.game.GameService$$EnhancerBySpringCGLIB$$ae68bc70.runWithReplay(<generated>)
	at com.faforever.client.replay.ReplayService.runFafReplayFile(ReplayService.java:373)
	at com.faforever.client.replay.ReplayService.runReplayFile(ReplayService.java:327)
	at com.faforever.client.replay.ReplayService$$FastClassBySpringCGLIB$$39329fdb.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
	at com.faforever.client.replay.ReplayService$$EnhancerBySpringCGLIB$$8c2adcc8.runReplayFile(<generated>)
	at com.faforever.client.os.FileOpeningHandler.runReplay(FileOpeningHandler.java:46)
	at com.faforever.client.os.FileOpeningHandler.run(FileOpeningHandler.java:56)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:768)
	... 7 more

```

**Screenshots**
![image](https://user-images.githubusercontent.com/38242291/162778939-c305ebbf-b911-44ba-ba3e-853325614123.png)

**OS**
Windows

--- a/src/main/java/com/faforever/client/os/FileOpeningHandler.java
+++ b/src/main/java/com/faforever/client/os/FileOpeningHandler.java
@@ -1,8 +1,13 @@
 package com.faforever.client.os;
 
+import com.faforever.client.fx.JavaFxUtil;
+import com.faforever.client.net.ConnectionState;
 import com.faforever.client.notification.NotificationService;
 import com.faforever.client.replay.ReplayService;
+import com.faforever.client.user.UserService;
 import com.install4j.api.launcher.StartupNotification;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.compress.compressors.CompressorException;
@@ -25,6 +30,7 @@ public class FileOpeningHandler implements ApplicationRunner, InitializingBean {
 
   private final ReplayService replayService;
   private final NotificationService notificationService;
+  private final UserService userService;
 
   @Override
   public void afterPropertiesSet() {
@@ -53,7 +59,16 @@ public class FileOpeningHandler implements ApplicationRunner, InitializingBean {
   public void run(ApplicationArguments args) {
     String[] sourceArgs = args.getSourceArgs();
     if (sourceArgs.length > 0) {
-      runReplay(Path.of(sourceArgs[0]));
+      ChangeListener<ConnectionState> connectionStateListener = new ChangeListener<>() {
+        @Override
+        public void changed(ObservableValue<? extends ConnectionState> observable, ConnectionState oldValue, ConnectionState newValue) {
+          if (newValue == ConnectionState.CONNECTED) {
+            JavaFxUtil.removeListener(userService.connectionStateProperty(), this);
+            runReplay(Path.of(sourceArgs[0]));
+          }
+        }
+      };
+      JavaFxUtil.addListener(userService.connectionStateProperty(), connectionStateListener);
     }
   }
 }

--- a/src/test/java/com/faforever/client/os/FileOpeningHandlerTest.java
+++ b/src/test/java/com/faforever/client/os/FileOpeningHandlerTest.java
@@ -1,8 +1,12 @@
 package com.faforever.client.os;
 
+import com.faforever.client.net.ConnectionState;
 import com.faforever.client.notification.NotificationService;
 import com.faforever.client.replay.ReplayService;
 import com.faforever.client.test.ServiceTest;
+import com.faforever.client.user.UserService;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
 import org.apache.commons.compress.compressors.CompressorException;
 import org.junit.jupiter.api.Test;
 import org.mockito.InjectMocks;
@@ -14,6 +18,8 @@ import java.io.IOException;
 import java.nio.file.Path;
 
 import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
 
 public class FileOpeningHandlerTest extends ServiceTest {
 
@@ -21,15 +27,22 @@ public class FileOpeningHandlerTest extends ServiceTest {
   private ReplayService replayService;
   @Mock
   private NotificationService notificationService;
+  @Mock
+  private UserService userService;
 
   @InjectMocks
   private FileOpeningHandler instance;
 
   @Test
-  public void run() throws IOException, CompressorException {
+  public void testRunLocalReplay() throws IOException, CompressorException {
+    ObjectProperty<ConnectionState> connectionState = new SimpleObjectProperty<>(ConnectionState.DISCONNECTED);
+    when(userService.connectionStateProperty()).thenReturn(connectionState);
+
     ApplicationArguments args = new DefaultApplicationArguments("foo.fafreplay");
     instance.run(args);
+    verifyNoInteractions(replayService);
 
+    connectionState.set(ConnectionState.CONNECTED);
     verify(replayService).runReplayFile(Path.of("foo.fafreplay"));
   }
 }

